connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/SparkResult.scala

```diff
@@ -46,7 +46,14 @@ private[sql] class SparkResult[T](
   private[this] var numRecords: Int = 0
   private[this] var structType: StructType = _
   private[this] var boundEncoder: ExpressionEncoder[T] = _
-  private[this] val batches = mutable.Buffer.empty[ColumnarBatch]
+  private[this] var nextBatchIndex: Int = 0
+  private[this] val idxToBatches = mutable.Map.empty[Int, ColumnarBatch]
+
+  // Exposed for UT.
+  private[sql] def existingBatches(): Seq[ColumnarBatch] = {
```

Comment:
Maybe we should use `PrivateMethodTester` to get `idxToBatches` in test

Reply:
thanks, done.